### PR TITLE
Features/pass params to emailer

### PR DESCRIPTION
**Pull request by wykhuh**
@patbenatar @taboularasa  Allow users to customize the email and flash success message by passing them as parameters.

### CODE REVIEW

File: app/mailers/report_card/mailer.rb [hunk 1]
```diff
@@ -1,11 +1,12 @@
 module ReportCard
   class Mailer < ActionMailer::Base
-    def report(url, email)
+    def report(url, email_options, email)
```
Comment (taboularasa) [inline app/mailers/report_card/mailer.rb]: why not put `email` in `email_options`?

File: README.md [hunk 2]
```diff
@@ -46,9 +46,15 @@ With reports defined in `app/reports`, you can navigate to
...
 ```ruby
 # config/initializers/report_card.rb
 ReportCard.parent_controller = 'Admin::BaseController'
 ReportCard.recipient_email = proc { current_admin.email }
 ReportCard.from_email = 'support@mysite.com'
-```
+ReportCard.subject = 'Your report is ready'
+ReportCard.body = 'Download your report: '
+ReportCard.flash_success = proc { 'Generating report. It will be emailed to you.' }
+```
```
Comment (taboularasa) [inline README.md]: missing newline

File: app/views/report_card/mailer/report.html.erb [hunk 3]
```diff
@@ -1 +1 @@
-Download your report: <%= link_to @url, @url %>
+<%= @body %> <%= link_to @url, @url %>
```
Comment (taboularasa) [inline app/views/report_card/mailer/report.html.erb]: missing newline

Comment (patbenatar) [inline app/views/report_card/mailer/report.html.erb]: We should use mustache templating here so the user can control where the link goes: `Your download is available here: {{link}} — thank you!` for instance

**Comment by taboularasa** [comment]
@wykhuh looks good overall, left some comments

File: app/mailers/report_card/mailer.rb (hunk 1 above)
Comment (patbenatar) [inline app/mailers/report_card/mailer.rb]: Should this just be an arbitrary `options` for use outside of the email use case later?

File: app/views/report_card/mailer/report.html.erb (hunk 3 above)
Comment (taboularasa) [inline app/views/report_card/mailer/report.html.erb]: interesting, would the template be passed by the client, or inferred by conventional location in the app directory?

Comment (patbenatar) [inline app/views/report_card/mailer/report.html.erb]: Yeah I was thinking about that.. could just use erb and put it in app/views/report_card/...

Comment (patbenatar) [inline app/views/report_card/mailer/report.html.erb]: I think if we're going to add support for customizing the template we should anticipate supporting more robust / HTML templates

File: README.md [hunk 4]
```diff
@@ -46,9 +46,15 @@ With reports defined in `app/reports`, you can navigate to
 
 ## Configuration
 
+Use proc instead of string  for recipient_email and flash_success in case they need to be evaluated.
+For instance, proc { current_admin.email }
```
Comment (patbenatar) [inline README.md]: Is it worth noting here that they will be evaluated within the controller instance? Also: use backticks for inline code blocks in markdown: "For instance, `proc { current_admin.email }`"

File: app/mailers/report_card/mailer.rb (hunk 1 above)
Comment (wykhuh) [inline app/mailers/report_card/mailer.rb]: I forgot change the code  to include `email` in the `email_options`.

File: app/views/report_card/mailer/report.html.erb (hunk 3 above)
Comment (wykhuh) [inline app/views/report_card/mailer/report.html.erb]: @patbenatar I'll add an issue for customizing the email template. This PR is blocking billable work.

File: app/controllers/report_card/reports_controller.rb [hunk 5]
```diff
@@ -6,11 +6,17 @@ def index
...
+          params[:email_options] = {}
+          params[:email_options][:recipient_email] = instance_eval(&ReportCard.recipient_email)
+        end
+
+        flash_success = params[:flash_success] || instance_eval(&ReportCard.flash_success)
+        ReportCard::Runner.perform_async(params)
+        redirect_to :back, flash: { success: flash_success }
       else
-        redirect_to report_card_reports_path, flash: { error: 'Could not find report' }
+        redirect_to :back, flash: { error: 'Could not find report' }
```
Comment (patbenatar) [inline app/controllers/report_card/reports_controller.rb]: :+1:

File: app/views/report_card/mailer/report.html.erb (hunk 3 above)
Comment (patbenatar) [inline app/views/report_card/mailer/report.html.erb]: @wykhuh Fair enough

**Comment by patbenatar** [comment]
@wykhuh Do you intend to release this build to rubygems or use off of master? It doesn't look like it breaks any backwards compatibility but can you confirm?

This is good to :eggplant: and :ship: ("squash commits and ship")

File: lib/report_card.rb [hunk 6]
```diff
@@ -7,8 +7,18 @@ module ReportCard
   self.parent_controller = 'ApplicationController'
 
   mattr_accessor :recipient_email
-  self.recipient_email = -> { 'change-me@example.com' }
+  self.recipient_email = proc { 'change-me@example.com' }
```
Comment (jdwolk) [inline lib/report_card.rb]: @wykhuh why this change?

File: lib/report_card.rb [hunk 7]
```diff
@@ -7,8 +7,18 @@ module ReportCard
...
+
+  mattr_accessor :subject
+  self.subject = 'Your report is ready'
+
+  mattr_accessor :body
+  self.body = 'Download your report: '
+
+  mattr_accessor :flash_success
+  self.flash_success = proc { 'Generating report. It will be emailed to you.' }
+
```
Comment (jdwolk) [inline lib/report_card.rb]: @wykhuh scrap the newline

File: lib/report_card.rb (hunk 6 above)
Comment (jdwolk) [inline lib/report_card.rb]: lol nvm, just saw https://github.com/philosophie/report_card/commit/a61ebd24e274a1ea429229607048dc60467bebb1#commitcomment-10880130

Comment (taboularasa) [inline lib/report_card.rb]: @jdwolk late to the party, @patbenatar @wykhuh and I already had a long convo on this topic
